presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardDao.java

```diff
@@ -80,6 +80,30 @@
     @Mapper(ShardMetadata.Mapper.class)
     Set<ShardMetadata> getNodeShards(@Bind("nodeIdentifier") String nodeIdentifier);
 
+    @SqlQuery("SELECT table_id, shard_id, shard_uuid, bucket_number, row_count, compressed_size, uncompressed_size\n" +
```

Comment:
This query is complex and I don't like to copy/paste it. See if we can do the trick where we add the optional filter like this: `(table_id = :tableId OR :tableId IS NULL)`

I hope the MySQL optimizer is smart enough to optimize that out since it's constant, but as we know that could be "unwarranted optimism". You'll need to run the query on the production database and compare the timing with and without the constant filter, i.e. `(table_id = NULL OR NULL IS NULL`)